forc/src/utils/helpers.rs

```diff
@@ -336,8 +337,16 @@ fn construct_window<'a>(
     let calculated_start_ix = calculated_start_ix.unwrap_or(0);
     let calculated_end_ix = calculated_end_ix.unwrap_or(input.len());
 
-    *start_ix -= calculated_start_ix;
-    *end_ix -= calculated_start_ix;
+    *start_ix = if *start_ix >= calculated_start_ix {
+        *start_ix - calculated_start_ix
+    } else {
+        0
+    };
+    *end_ix = if *end_ix >= calculated_start_ix {
```

Comment:
This could be a one-liner.  Code golf ahoy!
```
*start_ix -= std::cmp::min(calculated_start_ix, *start_ix);
```

Reply:
Love it! Changed, thanks.